yarn-project/simulator/src/avm/avm_gas.ts

```diff
@@ -133,123 +133,52 @@ const BaseGasCosts: Record<Opcode, Gas> = {
-  [Opcode.CHAINID]: makeCost(c.AVM_CHAINID_DYN_L2_GAS, 0),
-  [Opcode.VERSION]: makeCost(c.AVM_VERSION_DYN_L2_GAS, 0),
-  [Opcode.BLOCKNUMBER]: makeCost(c.AVM_BLOCKNUMBER_DYN_L2_GAS, 0),
-  [Opcode.TIMESTAMP]: makeCost(c.AVM_TIMESTAMP_DYN_L2_GAS, 0),
-  [Opcode.CALLDATACOPY]: makeCost(c.AVM_CALLDATACOPY_DYN_L2_GAS, 0),
-  [Opcode.L2GASLEFT]: makeCost(c.AVM_L2GASLEFT_DYN_L2_GAS, 0),
-  [Opcode.DAGASLEFT]: makeCost(c.AVM_DAGASLEFT_DYN_L2_GAS, 0),
-  [Opcode.JUMP_16]: makeCost(c.AVM_JUMP_DYN_L2_GAS, 0),
-  [Opcode.JUMPI_16]: makeCost(c.AVM_JUMPI_DYN_L2_GAS, 0),
-  [Opcode.INTERNALCALL]: makeCost(c.AVM_INTERNALCALL_DYN_L2_GAS, 0),
-  [Opcode.INTERNALRETURN]: makeCost(c.AVM_INTERNALRETURN_DYN_L2_GAS, 0),
-  [Opcode.SET_8]: makeCost(c.AVM_SET_DYN_L2_GAS, 0),
-  [Opcode.SET_16]: makeCost(c.AVM_SET_DYN_L2_GAS, 0),
-  [Opcode.SET_32]: makeCost(c.AVM_SET_DYN_L2_GAS, 0),
-  [Opcode.SET_64]: makeCost(c.AVM_SET_DYN_L2_GAS, 0),
-  [Opcode.SET_128]: makeCost(c.AVM_SET_DYN_L2_GAS, 0),
-  [Opcode.SET_FF]: makeCost(c.AVM_SET_DYN_L2_GAS, 0),
-  [Opcode.MOV_8]: makeCost(c.AVM_MOV_DYN_L2_GAS, 0),
-  [Opcode.MOV_16]: makeCost(c.AVM_MOV_DYN_L2_GAS, 0),
-  [Opcode.CMOV]: makeCost(c.AVM_CMOV_DYN_L2_GAS, 0),
-  [Opcode.SLOAD]: makeCost(c.AVM_SLOAD_DYN_L2_GAS, 0),
-  [Opcode.SSTORE]: makeCost(c.AVM_SSTORE_DYN_L2_GAS, 0),
-  [Opcode.NOTEHASHEXISTS]: makeCost(c.AVM_NOTEHASHEXISTS_DYN_L2_GAS, 0),
-  [Opcode.EMITNOTEHASH]: makeCost(c.AVM_EMITNOTEHASH_DYN_L2_GAS, 0),
-  [Opcode.NULLIFIEREXISTS]: makeCost(c.AVM_NULLIFIEREXISTS_DYN_L2_GAS, 0),
-  [Opcode.EMITNULLIFIER]: makeCost(c.AVM_EMITNULLIFIER_DYN_L2_GAS, 0),
-  [Opcode.L1TOL2MSGEXISTS]: makeCost(c.AVM_L1TOL2MSGEXISTS_DYN_L2_GAS, 0),
-  [Opcode.EMITUNENCRYPTEDLOG]: makeCost(c.AVM_EMITUNENCRYPTEDLOG_DYN_L2_GAS, 0),
-  [Opcode.SENDL2TOL1MSG]: makeCost(c.AVM_SENDL2TOL1MSG_DYN_L2_GAS, 0),
-  [Opcode.GETCONTRACTINSTANCE]: makeCost(c.AVM_GETCONTRACTINSTANCE_DYN_L2_GAS, 0),
-  [Opcode.CALL]: makeCost(c.AVM_CALL_DYN_L2_GAS, 0),
-  [Opcode.STATICCALL]: makeCost(c.AVM_STATICCALL_DYN_L2_GAS, 0),
-  [Opcode.DELEGATECALL]: makeCost(c.AVM_DELEGATECALL_DYN_L2_GAS, 0),
-  [Opcode.RETURN]: makeCost(c.AVM_RETURN_DYN_L2_GAS, 0),
-  [Opcode.REVERT_8]: makeCost(c.AVM_REVERT_DYN_L2_GAS, 0),
-  [Opcode.REVERT_16]: makeCost(c.AVM_REVERT_DYN_L2_GAS, 0),
-  [Opcode.DEBUGLOG]: makeCost(c.AVM_DEBUGLOG_DYN_L2_GAS, 0),
-  [Opcode.KECCAK]: makeCost(c.AVM_KECCAK_DYN_L2_GAS, 0),
-  [Opcode.POSEIDON2]: makeCost(c.AVM_POSEIDON2_DYN_L2_GAS, 0),
-  [Opcode.PEDERSEN]: makeCost(c.AVM_PEDERSEN_DYN_L2_GAS, 0),
-  [Opcode.ECADD]: makeCost(c.AVM_ECADD_DYN_L2_GAS, 0),
-  [Opcode.MSM]: makeCost(c.AVM_MSM_DYN_L2_GAS, 0),
-  [Opcode.PEDERSENCOMMITMENT]: makeCost(c.AVM_PEDERSENCOMMITMENT_DYN_L2_GAS, 0),
-  [Opcode.TORADIXLE]: makeCost(c.AVM_TORADIXLE_DYN_L2_GAS, 0),
-  [Opcode.SHA256COMPRESSION]: makeCost(c.AVM_SHA256COMPRESSION_DYN_L2_GAS, 0),
-  [Opcode.KECCAKF1600]: makeCost(c.AVM_KECCAKF1600_DYN_L2_GAS, 0),
-};
+const DYNAMIC_GAS_COSTS = new Map<Opcode, Gas>([
```

Comment:
You probably already realized but the advantage of using `Record<>` is that it forces you to cover all the cases. I'm ok with this since we don't often add opcodes with dynamic cases.